File: 55.jump-game/solution.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        if len(nums) == 1:
            return True
        j = 0
        for i in range(len(nums)):
            if j >= len(nums) - 1:
                return True
            if i > j:
                return False
            j = max(j, i + nums[i])

File: 55.jump-game/solution.go
func canJump(nums []int) bool {
	max := 0
	for i, n := range nums {
		if i > max {
			return false
		}
		if i == len(nums)-1 {
			break
		}
		if i+n > max {
			max = i + n
		}
	}
	return true
}
